lib/libc/common/source/stdio/remove.c

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (c) 2024 Linumiz.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+#include <stdio.h>
+#include <zephyr/fs/fs.h>
+
+/**
+ *
+ * @brief deletes a name from the filesystem
+ *
+ * @return On success, zero is returned. On error, -1 is returned
+ * and errno is set to indicate the error.
+ */
+
+int remove(const char *path)
+{
+	int ret;
+
```

Comment:
Something like this should work in case the FS is not enabled.

```suggestion
if (!IS_ENABLED(CONFIG_FS)) {
  errno = ENOTSUP;
  return -1;
}
```

Reply:
Yeah, but do the other part of remove in else. Otherwise you just carry the rest of function that will never be executed or start triggering static analysis that there is code after `if (1) { }`

Reply:
Dead code should be stripped by the compiler, so there should be no need for an explicit `else`